WordPressKit/WordPressComRestApi.swift

```diff
@@ -421,6 +422,100 @@ open class WordPressComRestApi: NSObject {
         let fileURL = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent(fileName)
         return fileURL
     }
+
+    // MARK: - Async
+
+    private lazy var urlSession: URLSession! = {
+        let configuration = URLSessionConfiguration.default
+
+        var additionalHeaders: [String: AnyObject] = [:]
+        if let oAuthToken = self.oAuthToken {
+            additionalHeaders["Authorization"] = "Bearer \(oAuthToken)" as AnyObject?
+        }
+        if let userAgent = self.userAgent {
+            additionalHeaders["User-Agent"] = userAgent as AnyObject?
+        }
+
+        configuration.httpAdditionalHeaders = additionalHeaders
+
+        return URLSession(configuration: configuration)
+    }()
+
+    // TODO: when migrating to Xcode 13.2 make this available to iOS 13
+    /**
+     Executes a GET request to the specified endpoint
+
+     - parameter url: the url `String` to be requested
+
+     - returns: a tuple containing the `Data` of the response and the `URLResponse` object
+     */
+    @available(iOS 15.0.0, *)
+    public func get(_ urlString: String) async throws -> (Data, URLResponse) {
+        guard let url = URL(string: urlString) else {
+            throw WordPressComRestApiError.malformedURL
+        }
+
+        return try await urlSession.data(from: url)
+    }
+
+    // TODO: when migrating to Xcode 13.2 make this available to iOS 13
+    /**
+     Executes a GET request to the specified endpoint
+
+     - parameter url: the url `String` to be requested
+     - parameter Type: an entity that conforms with `Codable`
+
+     - returns: a `Decodable` object of the `Type` given
+     */
+    @available(iOS 15.0.0, *)
+    public func get<T: Decodable>(_ urlString: String, ofType: T.Type) async throws -> T {
```

Comment:
Have you considered omitting the `ofType` parameter and having the caller help the compiler understand what `T` is?

What I'm suggesting is to modify the function signature to:

```suggestion
    public func get<T: Decodable>(_ urlString: String) async throws -> T {
```

And then call it by explicitly defining the type in the assignments, e.g. in the tests like:

```swift
let anyModel: AnyModel = try await api.get(url)
```

Personally, I find this option more appealing because it doesn't require an extra parameter.

Reply:
Weird! I swear I tried that and the compiler was giving me errors... I 100% prefer that too, changed.